### PR TITLE
Review Renderer + Layer refactor

### PR DESCRIPTION
**Pull request by VictorVelarde**
This PR has been elaborated during a final CR with David.

It includes 3 changes:
- add / remove some comments & typos in several classes.
- Small changes to `Renderer` moving some webgl sentences under an if based on geometry-type (supervised by David).
- `Layer` refactor: breaking some functions into little ones, specially `update`

### CODE REVIEW

File: src/renderer/Renderer.js [hunk 1]
```diff
@@ -268,13 +271,15 @@ export default class Renderer {
...
 
+        // Final drawing (to screen). In the case of lines / polygons, there is an extra step (for antialiasing)
         gl.blendFunc(gl.ONE, gl.ONE_MINUS_SRC_ALPHA);
         gl.enable(gl.BLEND);
 
-        gl.bindFramebuffer(gl.FRAMEBUFFER, null);
-        gl.viewport(0, 0, gl.drawingBufferWidth, gl.drawingBufferHeight);
-
-        if (renderLayer.type !== 'point') {
+        if (renderLayer.type === 'point') {
```
Comment (rochoa) [inline src/renderer/Renderer.js]: After https://github.com/CartoDB/carto-vl/pull/1146 gets merged, I guess we need to update this one with the master branch and reuse the constants.

File: src/renderer/Renderer.js [hunk 2]
```diff
@@ -345,21 +352,24 @@ export default class Renderer {
...
-            freeTexUnit++;
-
             gl.activeTexture(gl.TEXTURE0 + freeTexUnit);
             gl.bindTexture(gl.TEXTURE_2D, dataframe.texFilter);
             gl.uniform1i(renderer.filterTexture, freeTexUnit);
             freeTexUnit++;
-            gl.uniform2f(renderer.resolution, gl.canvas.width / window.devicePixelRatio, gl.canvas.height / window.devicePixelRatio);
+
+            // Specific Style textures
+            if (dataframe.type === 'point' || dataframe.type === 'line') {
```
Comment (rochoa) [inline src/renderer/Renderer.js]: Same here.

File: src/Layer.js [hunk 3]
```diff
@@ -66,18 +69,24 @@ export default class Layer {
...
 
+        this._initializeUIDS();
+        this._sourcePromise = this.update(source, viz);
+        this._renderWaiters = [];
+    }
+
+    /**
+     * Initialize unique identifiers to detect concurrent updates
+     */
+    _initializeUIDS () {
```
Comment (rochoa) [inline src/Layer.js]: I'm not a big fan of moving _initializers_ outside of the constructor.

File: src/renderer/Renderer.js (hunk 1 above)
Comment (VictorVelarde) [reply src/renderer/Renderer.js]: Great!, I'm gonna merge that one first and then adapt this!

File: src/Layer.js (hunk 3 above)
Comment (VictorVelarde) [reply src/Layer.js]: I'm not either. I had found the constructor a bit bloated and I was trying to reduce somehow its responsibilities. 

But your comment made me think about it and now I think I can refactor it a bit, extracting the UIDs mechanism. 👍

Comment (VictorVelarde) [reply src/Layer.js]: @rochoa check it please when you have a pair of minutes... 👇 commits

Comment (VictorVelarde) [reply src/Layer.js]: @rochoa you can check it if you want... (👇 commits). I'll merge this on Monday

File: src/LayerConcurrencyHelper.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+ * Helper to manage concurrent updates to a Layer.
+ *
+ * There are 2 type of changes in a layer's source or viz:
+ *      - Major changes. Major changes are performed by the `Layer.update` method and they will override (have priority over) minor changes
+ *      - Minor changes. Minor changes are performed by the `Layer.blendToViz` and `blendTo` method and they won't override concurrent calls to `Layer.update`
+ *
+ * @export
+ * @class LayerConcurrency
+ */
+export default class LayerConcurrencyHelper {
```
Comment (rochoa) [inline src/LayerConcurrencyHelper.js]: 👏 👏 👏

File: src/Layer.js (hunk 3 above)
Comment (rochoa) [reply src/Layer.js]: Much better, isn't it? 😄

**Pull request merged**